x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/dataframe/transforms/pivot/PivotConfigTests.java

```diff
@@ -136,6 +139,74 @@ public void testDoubleAggs() throws IOException {
         expectThrows(IllegalArgumentException.class, () -> createPivotConfigFromString(pivot, false));
     }
 
+    public void testAggNameValidations() throws IOException {
```

Comment:
tests are good, but I wonder if `aggFieldValidation(...)` could be re-factored in a way to test it at the unit test level with less boiler plate and more coverage?